### PR TITLE
Alcohol refactor v12413526345.

### PR DESCRIPTION
**Pull request by FlattestGuitar**
:cl: FlattestGuitar
tweak: changes some alcohol values, the weaker ones are no longer as weak
tweak: refactored alcohol, you can now take short breaks between sips and still pass out in a gentlemanly manner
tweak: different species have different levels of alcohol resistance, check in with your bartender for more information
/:cl:

*"No, Flattest!"* - the women cried - *"Don't do this, it's been enough!"*
But as he stood there, keyboard in hand, his mind wasn't focused on the wails of the opposition. His goal clearly visible, he was about to do what he should have done a long time ago.
Refactor alcohol code. Again.

Okay, so considering we're in the mood for self-serving PRs, this is my contribution. :smirk: 

I've moved almost all of alcohol code over to `human` rather than the reagent's `on_mob_life`, which allows for much smoother fade outs of being drunk. You will no longer simply stop being drunk as soon as whiskey leaves your system, you'll keep going for a long time, depending on how much you actually drank.

This also allows for much easier adminbus drunkness, because it's a simple variable that can get changed.

I've also taken the freedom of adding custom livers to most species, making them get hit by booze harder than Humans.

Full list:
- Dionae, Kida - 0.5x
- Drask, Unathi - 0.8x
- Vulpkanin, Tajaran - 1.4x
- Vox, Grey - 1.6x
- Skrell - 4x
- Everything else - stays at 1x, unless it has no liver

~~Finally, the buff that Humans deserve.~~ Why do you people hate humans so much?!

As it stands, because of my selfishness, the amount of alcohol you need to get drunk stay similar for tajaran and vulpkanin, the rest of species will get shifted higher or lower because of that. Experiment at the bar.

### CODE REVIEW

**Comment by taukausanake** [comment]
So skrell, the cryo aliums, and slimepeople have normal tolerance? I would imagine them to have the more diverse numbers. I would maybe imagine plasmamen to burn it out of their system faster too. I hope you're not done yet

**Comment by FlattestGuitar** [comment]
Skrell actually already have it times 5, the rest is still very much up for debate.

**Comment by FalseIncarnate** [comment]
As per lore, Kidan are nearly immune to the effects of alcohol, so the 2.5x is kinda wrong

**Comment by FlattestGuitar** [comment]
Hm, I'd love to see the lore remade, but you make a good point. 

On the other hand, humans should be able to prove their unwavering superiority to other species and this is the only objective way to do that. 

I think I'll tweak the amounts so that they're less wild, because the 2.5x does seem a bit overboard, plus I'd be breaking alcoholism for Khazar.

**Comment by FlattestGuitar** [comment]
Changed the liver effectiveness thing. Humans really need to be the best at emptying shot glasses.

**Comment by Spacemanspark** [comment]
Nah, not really.

**Comment by Da-Dman234** [comment]
Skrell masterrace

**Comment by KasparoVy** [comment]
>Be skrell
>Down one shot at the bar
>Floored :skull:

**Comment by FlattestGuitar** [comment]
@Fox-McCloud @TheDZD, can I get some opinion on this? The confused emote is scaring me.

**Comment by Spacemanspark** [comment]
I still think you should follow the lore and make put Kidan where they should be in terms of getting drunk.
Humans are 'vanilla', meaning they don't have any real downsides, but not many real upsides, either.

**Comment by Fox-McCloud** [comment]
I don't really understand or fully see why drunk is being made into its own entire process for human `life`, to be honest.

From what I can  gather, this doesn't seem to change much; the effects of drunkenness are going to stop occurring the moment the last bit of ethanol leaves your system (aside from the increased slurring and stuff like that, which is present on live as well)---only thing this *really* seems to do is make it so that once you stop drinking, you're in this state of not being drunk, but if you drink more, its intensity kicks back up faster.

**Comment by FlattestGuitar** [comment]
Yeah, I broke this in one of my commits. Fixed now.

The booze will keep affecting you after it leaves your system, that's the whole point here.

**Comment by Davdichan** [comment]
I dunno, humans being the the sturdiest ones seems... off. They aren't exactly the most durable race. Personally I'd think Unathi would be affected the least by booze.

Adding racial drink that affect different species differently would be interesting though.

**Comment by taukausanake** [comment]
Unathi getting the least drunk? I still think Drask or Kidan would be better choices for least drunk race. Humans should get basic immunity like the vanilla race they are

File: code/modules/mob/living/carbon/human/life.dm [hunk 1]
```diff
@@ -730,6 +731,80 @@
...
+				s.set_up(3, 1, src)
+				s.start()
+			if(alcohol_strength >= braindamage_start && prob(10))
+				adjustBrainLoss(1)
+
+	else
+		AdjustDrunk(-0.5)
+	return
+
+/mob/living/carbon/human/proc/has_booze() //checks if the human has ethanol or its subtypes inside
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/human/life.dm]: this proc is entirely unused.

**Comment by FlattestGuitar** [comment]
@Fox-McCloud, I forgot to switch that part out for `has_booze()` in the last commit, now it's fixed.

**Comment by FlattestGuitar** [comment]
@Fox-McCloud, switched all the whoopses to non-whoopses and adjusted the values for species. As ready as it gets from me.

**Comment by Necaladun** [comment]
While you're at it...can I request while you're at it that cubra libre requires lime as well? It just being rum and coke shits me. This is mainly because I actually did a 1st year anthro assignment about the cubra libre.

**Comment by KasparoVy** [comment]
Get the liquor quicker :')

**Pull request merged**